File: Berger.Data/MsfaEntity/Painter/Painter.cs
using Berger.Data.Common;
using Berger.Data.MsfaEntity.Setup;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Berger.Data.MsfaEntity.PainterRegistration
{
    [Table("Painters")]
    public class Painter :AuditableEntity<int>    {
        
        public Painter()
        {
            Attachments = new List<PainterAttachment>();
            AttachedDealers = new List<AttachedDealerPainter>();
            PainterCalls = new List<PainterCall>();
    }
        public string Depot { get; set; }
        public string SaleGroup { get; set; }
        public string Territory { get; set; }
        public string Zone { get; set; }
        public int PainterCatId { get; set; } // type of client
        public DropdownDetail PainterCat { get; set; } // type of client
        public string PainterName { get; set; }
        public string Address { get; set; }
        public string Phone { get; set; }
        public int NoOfPainterAttached { get; set; }
        public bool HasDbbl { get; set; }
        public string AccDbblNumber { get; set; }
        public string AccDbblHolderName { get; set; }
        public string PassportNo { get; set; }
        public string NationalIdNo { get; set; }
        public string BrithCertificateNo { get; set; }
        public string PainterImageUrl { get; set; }
        public string AttachedDealerCd { get; set; }
        public bool IsAppInstalled { get; set; }
        public string Remark { get; set; } // app not install reason
        public decimal AvgMonthlyVal { get; set; }
        public float Loyality { get; set; }
        public string EmployeeId { get; set; }

        public List<AttachedDealerPainter> AttachedDealers { get; set; } 
        public List<PainterAttachment> Attachments { get; set; }
        public List<PainterCall> PainterCalls { get; set; }

        public object Include(Func<object, object> p)
        {
            throw new NotImplementedException();
        }
    }

    public class AttachedDealerPainter: AuditableEntity<int>
    {
        public int Dealer { get; set; }
        public int PainterId { get; set; }

        [ForeignKey("PainterId")]
        public Painter Painter { get; set; }

    }



}

File: BergerMsfaApi/ClientApp/src/app/Shared/Entity/Painter/Painter.ts



export class Painter {

    public depotName: string;
    public painterName: string;
    public saleGroup: string;
    public Address: string;
    public Phone: string;
    public HasDbbl: boolean;
    public AccNumber: string;
    public AccHolderName: string;
    public PersonlIdentityNo: string;
    public painterImageUrl: string;
    public IsAppInstalled: boolean;
    public Remark: string;
    public AvgMonthlyVal: number;
    public Loality: number
    public DealerId: number;
    public PainterCatId: number;
    public TerritoryId: number;
    public status: number;
    public statusText: string;
    public statusBtnClass: string;
    public statusBtnIcon: string;
    

    constructor() {

    }
}

export class PainterUpdate {
    id: number;
    depot: string;
    // saleGroup: string;
    territory: string;
    zone: string;
    painterCatId: number;
    painterName: string;
    address: string;
    phone: string;
    noOfPainterAttached: number;
    hasDbbl: boolean;
    accDbblNumber: string;
    accDbblHolderName: string;
    passportNo: string;
    nationalIdNo: string;
    brithCertificateNo: string;
    painterImageUrl: string;
    painterImageBase64: string;
    // attachedDealerCd: string;
    isAppInstalled: boolean;
    remark: string;
    avgMonthlyVal: number;
    loyality: number;
    employeeId: string;
    painterNo: string;
    painterCode: string;
    attachedDealerIds: number[];

    constructor(init?: Partial<PainterUpdate>) {
        this.attachedDealerIds = [];
        Object.assign(this, init);
    }

    clear() {
        this.attachedDealerIds = [];
    }
}